CefSharp.Wpf/ChromiumWebBrowser.cs

```diff
@@ -960,7 +964,47 @@ void IRenderWebBrowser.OnCursorChange(IntPtr handle, CursorType type, CursorInfo
 
         void IRenderWebBrowser.OnImeCompositionRangeChanged(Range selectedRange, Rect[] characterBounds)
         {
-            OnImeCompositionRangeChanged(selectedRange, characterBounds);
+            Visual GetParentWindow()
+            {
+                var current = VisualTreeHelper.GetParent(this);
+                while (current != null && !(current is Window))
+                {
+                    current = VisualTreeHelper.GetParent(current);
+                }
+
+                return current as Window;
+            }
+
+            var imeKeyboardHandler = WpfKeyboardHandler as WpfKeyboardHandler;
+            if (imeKeyboardHandler.IsActive)
+            {
+                var screenInfo = GetScreenInfo();
+                var scaleFactor = screenInfo.HasValue ? screenInfo.Value.DeviceScaleFactor : 1.0f;
+
+                UiThreadRunSync(() =>
```

Comment:
Is there a reason this is run sync? Add comment if there is.

Reply:
We are manually tracking screen location already using https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Wpf/ChromiumWebBrowser.cs#L1741
Should be able to reuse that and do away with the sync call.

Reply:
GetBrowserScreenLocation() returns absolute position, here is used relative position.

Reply:
Please add a detailed description of what is being calculated and the reason for it, comment you added above is pretty meaningless. 

Relative position to what? At a minimum this will need to be run asynchronous, running sync is too problematic,